lib/utils/setDeclarationValue.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+const _ = require('lodash');
+
+/** @typedef {import('postcss').Declaration} Declaration */
+
+/**
+ * @param {Declaration} decl
+ * @param {string} value
+ * @returns {Declaration} The declaration that was passed in.
+ */
+module.exports = function getDeclarationValue(decl, value) {
+	if (_.has(decl, 'raws.value')) {
+		_.set(decl, 'raws.value.raw', value);
```

Comment:
This change uses lodash because `Declaration` doesn't strictly adhere to the version of `postcss.Declaration` that we use. `stylelint` uses PostCSS 7 and, as best as I can tell, these utilities were written so that support for migration to PostCSS 8 would be easier? Or perhaps the typescript definitions in PostCSS 7 are simply wrong.

The [`Declaration`](https://github.com/postcss/postcss/blob/12832f3d203474bd273bd06bd3b2407567bfe09e/lib/postcss.d.ts#L1195-L1222) interface in PostCSS 7 inherits from [`NodeBase`](https://github.com/postcss/postcss/blob/12832f3d203474bd273bd06bd3b2407567bfe09e/lib/postcss.d.ts#L588-L707), where [`raws`](https://github.com/postcss/postcss/blob/12832f3d203474bd273bd06bd3b2407567bfe09e/lib/postcss.d.ts#L599-L603) is defined as [`NodeRaws`](https://github.com/postcss/postcss/blob/12832f3d203474bd273bd06bd3b2407567bfe09e/lib/postcss.d.ts#L712-L748) and does not support a `value` property.

In PostCSS 8, there is a [`declaration.d.ts`](https://github.com/postcss/postcss/blob/4f4e2932fc97e2c117e1a4b15f0272ed551ed59d/lib/declaration.d.ts) file that has a [`DeclarationRaws`](https://github.com/postcss/postcss/blob/4f4e2932fc97e2c117e1a4b15f0272ed551ed59d/lib/declaration.d.ts#L3-L27) interface that *does* supply a type signature that meets the first definition, however.

I thought about creating some complicated `@typedefs`, but it seemed like we were already using lodash in a few places to get around this type signature problem. Let me know if you have any suggestions on how to define this better, though.

Reply:
Great! 👏 
I think it would be nice if you could leave the reason as a code comment as below:

```js
// The reason to use Lodash here: ...
// See also: https://github.com/stylelint/stylelint/pull/5183/files#r588047494
if (_.has(decl, 'raws.value')) {
```